app/services/pii/cacher.rb

```diff
@@ -31,16 +31,32 @@ def fetch
       Pii::Attributes.new_from_json(pii_string)
     end
 
+    # Between requests, the decrypted PII bundle is encrypted with KMS and moved to the
+    # 'encrypted_pii' key by the SessionEncryptor.
+    #
+    # The PII is decrypted on-demand by this method and moved into the 'decrypted_pii' key.
+    # See SessionEncryptor#kms_encrypt_pii! for more detail.
     def fetch_string
-      user_session[:decrypted_pii]
+      return unless user_session[:decrypted_pii] || user_session[:encrypted_pii]
+      if user_session[:decrypted_pii]
+        user_session[:decrypted_pii]
+      elsif user_session[:encrypted_pii]
+        decrypted = SessionEncryptor.new.kms_decrypt(
+          user_session[:encrypted_pii],
+        )
+        user_session[:decrypted_pii] = decrypted
+
+        decrypted
```

Comment:
**question:** wouldn't this be returned automatically from line 47? Totally fine if you prefer an explicit return value anyway

Reply:
it does, yeah. I have a slight preference towards the explicit return in terms of readability, but I'm open to changing it